src/nvim/regexp_nfa.c

```diff
@@ -4853,6 +4854,9 @@ static int nfa_regmatch(nfa_regprog_T *prog, nfa_state_T *start, regsubs_T *subm
   fast_breakcheck();
   if (got_int)
     return FALSE;
+  if (nfa_time_limit != NULL && profile_passed_limit(*nfa_time_limit)) {
+    return FALSE;
```

Comment:
This is flagging a clint error.  The return is really a tri-state (TRUE, FALSE, NFA_TOO_EXPENSIVE).